bin/cron/pardot_import

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env ruby
+
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require 'net/http'
+require 'net/http/responses'
+require_relative('../../dashboard/config/environment')
+require 'cdo/properties'
+
+# Global variable for Pardot API key. This can become invalid and need to be refreshed
+# and replaced midstream.
+$api_key = nil
+
+# Exception to throw to ourselves if Pardot API key is invalid (which probably means it needs to be
+# re-authed)
+class InvalidApiKeyException < Exception
+end
+
+# URL for Pardot login
+PARDOT_AUTHENTICATION_URL = "https://pi.pardot.com/api/login/version/3"
+# URL for prospect batch creation
+PARDOT_BATCH_CREATE_URL = "https://pi.pardot.com/api/prospect/version/4/do/batchCreate"
+# URL for prospect single creation
+PARDOT_CREATE_PROSPECT_URL = "https://pi.pardot.com/api/prospect/version/4/do/create/email"
+
+# Max # of prospects allowed in one batch
+MAX_PROSPECT_BATCH_SIZE = 25
+# Emperically determined safe max URL length for batch operations
+MAX_URL_LENGTH = 6200
+
+# Map of database fields to Pardot field names
+MYSQL_TO_PARDOT_MAP = {
+    email: { field: 'email', multi: false},
+    name: { field: 'first_name', multi: false},
+    street_address: { field: 'address_one', multi: false},
+    city: { field: 'city', multi: false},
+    state: { field: 'state', multi: false},
+    postal_code: { field: 'zip', multi: false},
+    country: { field: 'country', multi: false},
+    district_name: { field: 'District', multi: false},
+    district_city: { field: 'District_City', multi: false},
+    district_state: { field: 'District_State', multi: false},
+    school_name: { field: 'School', multi: false},
+    roles: { field: 'Roles', multi: true},
+    courses_facilitated: { field: 'Facilitator_Type', multi: true},
+    professional_learning_enrolled: { field: 'Professional_Learning_Enrolled', multi: true},
+    professional_learning_attended: { field: 'Professional_Learning_Attended', multi: true},
+    hoc_organizer_years: { field: 'Hour_of_Code_Organizer', multi: true},
+    grades_taught: { field: 'Grades_Taught', multi: true},
+    ages_taught: { field: 'Ages_Taught', multi: true}
+}
+
+# Log into Pardot and request an API key. The API key is valid for (up to) one hour, after which
+# it will become invalid and we will need to request a new one.
+def request_pardot_api_key
+  uri = URI.parse(PARDOT_AUTHENTICATION_URL)
+
+  puts "Requesting new API key"
```

Comment:
Yes. Have moved all of these into a log method which for the moment saves to log and emits to console. In the near term I will be running interactively and need to see output. As I move it to real cron job I will remove console output.